src/components/AutoUpdateTime.js

```diff
@@ -0,0 +1,86 @@
+import {View} from 'react-native';
+import React, {PureComponent} from 'react';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+import DateUtils from '../libs/DateUtils';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import Timers from '../libs/Timers';
+import Text from './Text';
+
+const propTypes = {
+    timezone: PropTypes.shape({
+    // Value of selected timezone
+        selected: PropTypes.string,
+
+        // Whether timezone is automatically set
+        automatic: PropTypes.bool,
+    }),
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    timezone: null,
+};
+
+class AutoUpdateTime extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.getCurrentUserLocalTime = this.getCurrentUserLocalTime.bind(this);
+        this.state = {
+            timezone: this.getCurrentUserLocalTime(),
+        };
+    }
+
+    componentDidMount() {
+        this.timer = Timers.register(setInterval(() => {
+            this.setState({
+                timezone: this.getCurrentUserLocalTime(),
+            });
+        }, 60000));
+    }
+
+    componentWillUnmount() {
+        clearInterval(this.timer);
+    }
+
+    getCurrentUserLocalTime() {
+        // if timezone isn't set, use local time instead.
+        return this.props.timezone
+            ? DateUtils.getLocalMomentFromDatetime(
+                this.props.preferredLocale,
+                null,
+                this.props.timezone.selected,
+            )
+            : moment();
```

Comment:
This is hard to read as it's written, so let's use a fail-fast pattern here
```suggestion
        if(!this.props.timezone) {
            return moment();
        }
        return DateUtils.getLocalMomentFromDatetime(this.props.preferredLocale,null, this.props.timezone.selected);
```
Also is we don't have `this.props.timezone`, why do we need to return `moment()` instead of null here? in this case the date won't be shown right? Given that, can we make the timezone prop required?